drivers/adc/adc_mcux_lpadc.c

```diff
@@ -57,7 +58,65 @@ struct mcux_lpadc_data {
 	lpadc_conv_command_config_t cmd_config[CONFIG_LPADC_CHANNEL_COUNT];
 };
 
+static int mcux_lpadc_acquisition_time_setup(const struct device *dev, uint16_t acq_time,
+					     lpadc_conv_command_config_t *cmd)
+{
+	const struct mcux_lpadc_config *config = dev->config;
+	bool find_suitable_option = false;
+	uint32_t adc_freq_hz = 0;
+	uint32_t adc_cycle_ns = 0;
+	uint32_t acquisition_time_value = ADC_ACQ_TIME_VALUE(acq_time);
+	uint8_t acquisition_time_unit = ADC_ACQ_TIME_UNIT(acq_time);
+	uint32_t sample_time_array[] = {3, 5, 7, 11, 19, 35, 67, 131};
+	uint8_t sample_time_array_size = ARRAY_SIZE(sample_time_array);
+
+	if (acquisition_time_value == ADC_ACQ_TIME_DEFAULT) {
+		return 0;
+	}
 
+	if (ADC_ACQ_TIME_TICKS == acquisition_time_unit) {
+		for (uint8_t index = 0; index < sample_time_array_size; ++index) {
+			if (sample_time_array[index] >= acquisition_time_value) {
+				cmd->sampleTimeMode = (lpadc_sample_time_mode_t)index;
+				find_suitable_option = true;
+				break;
+			}
+		}
+
+		if (!find_suitable_option) {
+			return -EINVAL;
+		}
+	} else {
+		if (clock_control_get_rate(config->clock_dev, config->clock_subsys, &adc_freq_hz)) {
+			LOG_ERR("Get clock rate failed");
+			return -EINVAL;
+		}
+
+		adc_cycle_ns = (uint32_t)((float)1000000000.0 / adc_freq_hz);
+
+		for (uint8_t index = 0; index < sample_time_array_size; ++index) {
+			sample_time_array[index] = sample_time_array[index] * adc_cycle_ns;
+		}
+
+		if (ADC_ACQ_TIME_MICROSECONDS == acquisition_time_unit) {
+			acquisition_time_value *= 1000;
+		}
+
+		for (uint8_t index = 0; index < sample_time_array_size; ++index) {
+			if (sample_time_array[index] >= acquisition_time_value) {
+				cmd->sampleTimeMode = (lpadc_sample_time_mode_t)index;
+				find_suitable_option = true;
+				break;
+			}
+		}
```

Comment:
rather than using the stack array and the for loops, can't you just have a if/else if code and a multiplier variable?

Reply:
a loop brings extra code size too though, and these stack variables are also in the flash since they are literals, so that's not much different as far as code size, the issue is the stack size when having an array, the stack sizes on the embedded devices using zephyr will be limited and we should reduce the use of it

Reply:
I feel confused about the **these stack variables are also in the flash since they are literals**, I checked the .map file, the stack is located in the RAM area. 

You are right, we should really pay attention to the use of stacks in embedded devices. But I think there is no big problem with the stack used in the code added by this PR, because this is a NOT big array used during the initialization process, this stack will also be released at the end of the function call, and the function call depth is very low.

My main consideration is that I don’t want to write code that is too hardcoded (directly embedding values ​​such as 3, 5, ... into the code), which will make it very troublesome to maintain the code later.

Reply:
I will update the code with removing the array and using the if/else code instead of using for loop.